quipucords/scanner/network/processing/installed_products.py

```diff
@@ -0,0 +1,34 @@
+"""Initial processing of the shell output from the installed_products role."""
+
+import logging
+
+from scanner.network.processing import process
+
+logger = logging.getLogger(__name__)
+
+
+class ProcessInstalledProducts(process.Processor):
+    """Process the installed_products fact."""
+
+    KEY = "installed_products"
+
+    @staticmethod
+    def process(output, dependencies=None):
+        """Process installed_product fact output."""
+        products = []
+        installed_products_cmd_output = output.get("stdout_lines", [])
+        for index, line in enumerate(installed_products_cmd_output):
+            if line.strip().startswith("ID:"):
+                _, id_value = line.strip().split(":", 1)
+                id_value = id_value.strip()
+                if index + 1 < len(installed_products_cmd_output):
+                    name_line = installed_products_cmd_output[index + 1]
+                    if name_line.strip().startswith("Name:"):
```

Comment:
Why is the Name: check in an index conditional (line 24), isnt't just another possible line Syntax, so this check is a peer to line 21 (check for ID:) ?

Reply:
If I understood your questions correctly, the name by itself is not relevant for HBI, therefore I will only collect it if an ID is associated with it.

Reply:
👍